packages/static_shock/lib/src/static_shock.dart

```diff
@@ -484,31 +514,36 @@ class StaticShock implements StaticShockPipeline {
 
     // Load local layouts and components. We do this after loading remove values so
     // that local values can overwrite remote values.
-    for (final sourceFile in _sourceFiles.layouts()) {
-      _log.detail("Layout: ${sourceFile.subPath}");
-      _context.putLayout(
-        Layout(
-          FileRelativePath.parse(sourceFile.subPath),
-          sourceFile.file.readAsStringSync(),
-        ),
-      );
-    }
-    for (final sourceFile in _sourceFiles.components()) {
-      _log.detail("Component: ${sourceFile.subPath}");
-
-      final componentContent = front_matter.parse(sourceFile.file.readAsStringSync());
-
-      // TODO: process the component data, e.g., pull out CSS imports
-      _context.putComponent(
-        path.basenameWithoutExtension(sourceFile.subPath),
-        Component(
-          FileRelativePath.parse(sourceFile.subPath),
-          Map.from(componentContent.data),
-          // If there's no Front Matter, then `content` will be `null`. In that case, assume
-          // everything is a Jinja template, and pass the full `value`.
-          componentContent.content ?? componentContent.value,
-        ),
-      );
+    final sourceSets = {_sourceFiles, ..._sourceExtensions};
```

Comment:
These should be ordered from extensions to source files so source files can overwrite extensions. We already do this in `_pickAllSourceFiles()`. We should probably create a central method that returns all source sets in that priority order.